tests/python/unittest/test_gluon_rnn.py

```diff
@@ -152,190 +124,7 @@ def test_lstmp():
         check_rnn_states(fused_states, stack_states, num_layers, True)
 
 
-@assert_raises_cudnn_not_satisfied(min_version='5.1.10')
-def test_lstm_cpu_inference():
-    # should behave the same as lstm cell
-    EXPECTED_LSTM_OUTPUT = np.array([[[0.72045636, 0.72045636, 0.95215213, 0.95215213],
-                                      [0.72045636, 0.72045636, 0.95215213, 0.95215213]],
-                                     [[0.95215213, 0.95215213, 0.72045636, 0.72045636],
-                                      [0.95215213, 0.95215213, 0.72045636, 0.72045636]]])
-    x = mx.nd.ones(shape=(2, 2, 2))
-    model = mx.gluon.rnn.LSTM(2, num_layers=6, bidirectional=True)
-    model.initialize(mx.init.One())
-
-    y = model(x).asnumpy()
-    mx.test_utils.assert_almost_equal(y, EXPECTED_LSTM_OUTPUT,
-                                      rtol=1e-3, atol=1e-5)
-
-
-def test_gru():
-    cell = gluon.rnn.GRUCell(100, activation='relu', recurrent_activation='tanh')
-    inputs = [mx.sym.Variable('t%d_data'%i) for i in range(3)]
-    outputs, _ = cell.unroll(3, inputs)
-    outputs = mx.sym.Group(outputs)
-    assert sorted(cell.collect_params().keys()) == ['h2h_bias', 'h2h_weight', 'i2h_bias', 'i2h_weight']
-    assert outputs.list_outputs() == [
-        'grucell_t0_out_output', 'grucell_t1_out_output',
-        'grucell_t2_out_output'
-    ]
-
-    args, outs, auxs = outputs.infer_shape(t0_data=(10,50), t1_data=(10,50), t2_data=(10,50))
-    assert outs == [(10, 100), (10, 100), (10, 100)]
-
-
-@pytest.mark.serial
-def test_residual():
```

Comment:
Should this test be adapted to testing the front-end infer_shape functions instead?